---
title: "Genetically Proxied Lipid Therapy And Sepsis"
author: "Nicholas Sunderland"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Genetically Proxied Lipid Therapy And Sepsis}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
  wrap: 80
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Setup
First we need to load the MRLipidInfection package and setup package requirements by calling
'setupMRLipidInfection()'. At the first installation this will take some time as this will download
the 1000 genomes LD reference panel into the package file structure and ensure that PLINK is
installed locally.

```{r setup, eval=TRUE}
library(MRLipidInfection)
library(dplyr)

setupMRLipidInfection()
```
## Exposure data
Next we must obtain the data for the exposure. We are interested whether exposure to lipid lowering
medications influences risk of developing sepsis. This data is not readily available on a population
level and has many confounding factors. We will therefore estimate the effect of genetically proxied
lipid medication therapy by looking for Single Nucleotide Polymorphisms (SNPs) in close proximity to
known lipid metabolism genes (e.g. HMG-CoA reductase) that are significantly associated with
circulating lipid levels. We will use the the latest Global Lipids Genetics Consortium
[Ancestry-specific GWAS summary statistics](http://csg.sph.umich.edu/willer/public/glgc-lipids2021/)
for HDL-C, LDL-C, nonHDL-C, TC and TG. 

First we will download the data for LDL cholesterol - **Warning** GWAS data file is large and will
take some time to download.

```{r, eval=TRUE}
# The URL for the GWAS summary statistics README document
url_1 <- "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/README.txt"

# Download README to the package downloads directory (subdirectory "glgc").
get_web_file(url_1, directory="glgc")

# The URL for the GWAS summary statistics data
url_2 <- "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"

# Download '.gz' file to the package downloads directory (subdirectory "glgc").
fp_2 <- get_web_file(url_2, directory="glgc")

# GWAS data file now at:
fp_2
```

There are more than 47 million SNPs in the dataset. We now filter the GWAS data SNPs associated with
circulating LDL cholesterol levels at a statisticsal significance of <5e-8. Reading this file can
take a long time and so we can write out the result to a .csv file.

```{r, eval=TRUE}
# Read the .gz data file
d <- data.table::fread(fp_2) |>

  # Ensure that the pvalue is numeric 
  dplyr::mutate(pvalue = as.numeric(.data$pvalue)) |>

  # Filter for SNPs with a pvalue <5e-8
  dplyr::filter(.data$pvalue < 5e-8) |>
  
  # Ensure unique SNPs
  dplyr::distinct(.data$rsID, .keep_all=TRUE)

utils::head(d)

# The path to write to (place it alongside the main data file in the downloads directory)
fp_d <- paste0(get_downloads_dir(subdir="glgc"), "/glgc_sig_snps.csv")

# Write out the significant SNPs
readr::write_csv(d, fp_d)
```

In the case we want to re-run the analysis at any point we can just read back in the smaller .csv
file.

```{r, eval=TRUE}
# The path to the significant exposure SNPs
fp_d <- paste0(get_downloads_dir(subdir="glgc"), "/glgc_sig_snps.csv")

# Read back in
d <- readr::read_csv(fp_d, show_col_types=F)
```

If we inspect the column names we see that they need some adjustment to ensure they are compatible
with the TwoSampleMR package naming conventions.
```{r, eval=TRUE}
# The current column names
colnames(d)

# Standardise the exposure data column names (set clump to FALSE as we will do this later)
exposure_data <- TwoSampleMR::read_exposure_data(
    filename          = fp_d,
    clump             = FALSE,
    sep               = ",",
    snp_col           = "rsID",
    beta_col          = "EFFECT_SIZE",
    se_col            = "SE",
    eaf_col           = "POOLED_ALT_AF",
    effect_allele_col = "ALT",
    other_allele_col  = "REF",
    pval_col          = "pvalue",
    samplesize_col    = "N",
    chr_col           = "CHROM",
    pos_col           = "POS_b37")

# Add 'LDL' as the exposure tag
exposure_data$exposure <- "LDL"
```
