---
title: "Genetically Proxied Lipid Therapy And Sepsis"
author: "Nicholas Sunderland"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Genetically Proxied Lipid Therapy And Sepsis}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
  wrap: 80
---

```{r init, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Installation
You can install the latest version of MRLipidInfection in R using:
```{r installation, eval=FALSE}
remotes::install_github("nicksunderland/MRLipidInfection")
```

## Setup
First we need to load the MRLipidInfection package and setup package requirements by calling
'setupMRLipidInfection()'. At the first installation this will take some time as this will download
the [1000 genomes LD reference panel](http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz) into the package
file structure and ensure that [PLINK](https://www.cog-genomics.org/plink/) is installed locally.

```{r setup, eval=FALSE}
library(MRLipidInfection)

setupMRLipidInfection()
```

# Exposure data
Next we must obtain the data for the exposure. We are interested whether exposure to lipid lowering
medications influences risk of developing sepsis. This data is not readily available on a population
level and has many confounding factors. We will therefore estimate the effect of genetically proxied
lipid medication therapy by looking for Single Nucleotide Polymorphisms (SNPs) in close proximity to
known lipid metabolism genes (e.g. HMG-CoA reductase) that are significantly associated with
circulating lipid levels. We will use the the latest Global Lipids Genetics Consortium
[Ancestry-specific GWAS summary statistics](http://csg.sph.umich.edu/willer/public/glgc-lipids2021/)
for HDL-C, LDL-C, nonHDL-C, TC and TG. 

First we will download the data for LDL cholesterol - **Warning** GWAS data file is large and will
take some time to download.

```{r download_data, eval=FALSE}
# The URL for the GWAS summary statistics README document
url_1 <- "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/README.txt"

# Download README to the package downloads directory (subdirectory "glgc").
get_web_file(url_1, directory="glgc")

# The URL for the GWAS summary statistics data
url_2 <- "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"

# Download '.gz' file to the package downloads directory (subdirectory "glgc").
fp_2 <- get_web_file(url_2, directory="glgc")

# GWAS data file now at:
fp_2
```

There are more than 47 million SNPs in the dataset. We now filter the GWAS data SNPs associated with
circulating LDL cholesterol levels at a statisticsal significance of <5e-8. Reading this file can
take a long time and so we can write out the result to a .csv file.

```{r significant_snps, eval=FALSE}
# Read the .gz data file
d <- data.table::fread(fp_2) |>

  # Ensure that the pvalue is numeric 
  dplyr::mutate(pvalue = as.numeric(.data$pvalue)) |>

  # Filter for SNPs with a pvalue <5e-8
  dplyr::filter(.data$pvalue < 5e-8) |>
  
  # Ensure unique SNPs
  dplyr::distinct(.data$rsID, .keep_all=TRUE)

utils::head(d)

# The path to write to (place it alongside the main data file in the downloads directory)
fp_d <- paste0(get_downloads_dir(subdir="glgc"), "/glgc_sig_snps.csv")

# Write out the significant SNPs
readr::write_csv(d, fp_d)
```

In the case we want to re-run the analysis at any point we can just read back in the smaller .csv
file.
```{r read_data, eval=FALSE}
# The path to the significant exposure SNPs
fp_d <- paste0(get_downloads_dir(subdir="glgc"), "/glgc_sig_snps.csv")

# Read back in
d <- readr::read_csv(fp_d, show_col_types=F)
```

If we inspect the column names we see that they need some adjustment to ensure they are compatible
with the TwoSampleMR package naming conventions.
```{r standardise, eval=FALSE}
# The current column names
colnames(d)

# Standardise the exposure data column names (set clump to FALSE as we will do this later)
exposure_data <- TwoSampleMR::read_exposure_data(
    filename          = fp_d,
    clump             = FALSE,
    sep               = ",",
    snp_col           = "rsID",
    beta_col          = "EFFECT_SIZE",
    se_col            = "SE",
    eaf_col           = "POOLED_ALT_AF",
    effect_allele_col = "ALT",
    other_allele_col  = "REF",
    pval_col          = "pvalue",
    samplesize_col    = "N",
    chr_col           = "CHROM",
    pos_col           = "POS_b37")

# Add 'LDL' as the exposure tag
exposure_data$exposure <- "LDL"
```

## Cis-acting HMG-CoA Redictase SNPs
We will create a second set of exposure SNPs that will be subset of the SNPs that associate with LDL
cholesterol levels. This subset of SNPs will be within a certain proximity (number of bases) to the
HMG-CoA Reductatase gene. Data regarding the position of the HMG-CoA-R gene is freely available
[online](https://www.genecards.org/cgi-bin/carddisp.pl?gene=HMGCR). The Lipid data from the GLGC
uses the Genome Reference Consortium Human Reference 37 (GRCh37) and so we must use this when
referring to the position of the HMG-CoA-R gene.

```{r cis_hmg, eval=FALSE}
# The HMG-CoA-R gene is on chromosome 5, starts at base 746,32,993 and ends at base 74,657,941.
chrom <- 5
start <- 74632993
end   <- 74657941

# We want to capture SNPs near the HMG-CoA gene and so we must set a tolerance window either side in
# which to also search
tol <- 3e5

# Create a new dataset of SNPs close to the HMG-CoA gene
cis_hmg_exp_dat <- exposure_data |>
  dplyr::filter(.data$chr.exposure == chrom,
                .data$pos.exposure > (start - tol),
                .data$pos.exposure < (end + tol))
  
# 
```

## Clumping lipid exposure SNPs
Many of the SNPs will be highly correlated, i.e. be genetically linked and provide no additional
information over and above the whole set of correlated SNPs. To reduce the number of SNPs to only
those mopst significantly associated with lipid levels we use Linkage Disequilibrium clumping.
[Clumping](https://kp4cd.org/sites/default/files/documents/LD_clumping.pdf) reports the most
significant genetic associations in a region in terms of a smaller number of “clumps” of genetically
linked SNPs.

The clumping procedure has several parameters:
`clump_kb`
`clump_r2`
`clump_p`
We will use the defaults except for `clump_r2` which we will set to 0.1 instead
of the default of 0.001. This will clump SNPs that are correlated with an R^2 correlation of >0.1 -
the SNP with the lower p-value will be removed.
```{r clumping_params, eval=FALSE}
clump_r2 <- 0.1
```

We could do this on the Bristol IEU severs using the function `TwoSampleMR::clump_data`. 
```{r clumping_1, eval=FALSE}
clumped_exp <- exposure_data |> 
  TwoSampleMR::clump_data(clump_r2 = clump_r2)
```

Alternatively, we can do this locally using the 1000 Genomes Reference Panel and PLINK binary that
we downloaded earlier.
```{r clumping, eval=FALSE}
clumped_exp <- ieugwasr::ld_clump(
  dat       = dplyr::tibble(rsid=exposure_data$SNP, pval=exposure_data$pval.exposure),
  plink_bin = genetics.binaRies::get_plink_binary(),
  bfile     = get_b_file(), 
  clump_r2  = clump_r2
)
```
# Outcome data
Sepsis GWAS...

## To be continued
