---
title: "Genetically Proxied Lipid Therapy And Sepsis"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Genetically Proxied Lipid Therapy And Sepsis}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
  wrap: 80
---

```{r init, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment  = "#>", 
  message  = FALSE,
  warning  = FALSE, 
  echo     = TRUE,
  #eval     = TRUE,
  cache    = FALSE
)
```

## Installation
You can install the latest version of MRLipidInfection in R using:
```{r installation, eval = FALSE}
remotes::install_github("nicksunderland/MRLipidInfection")
```

## Setup
First we need to load the MRLipidInfection package and setup package requirements by calling
`setupMRLipidInfection()`. At the first installation this will take some time as this will download
the [1000 genomes LD reference panel](http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz) into the package
file structure and ensure that [PLINK](https://www.cog-genomics.org/plink/) is installed locally. If
you plan to use only the IEU servers for LD processing you can skip these downloads.

```{r setup, eval = FALSE}
library(MRLipidInfection)

setupMRLipidInfection(skip_local_download=TRUE)
```

<br>

# Exposure data
## Lipid profile data
Next we must obtain the data for the exposure. We are interested whether exposure to lipid lowering
medications influences risk of developing sepsis. This data is not readily available on a population
level and has many confounding factors. We will therefore estimate the effect of genetically proxied
lipid medication therapy by looking for Single Nucleotide Polymorphisms (SNPs) in close proximity to
known lipid metabolism genes (HMG-CoA Reductase, NPC1L1, PCSK9) that are significantly associated
with circulating lipid levels. We will use the the latest Global Lipids Genetics Consortium
[Ancestry-specific GWAS summary statistics](http://csg.sph.umich.edu/willer/public/glgc-lipids2021/)
for HDL-C, LDL-C, nonHDL-C, TC and TG. As we will be using outcome data from the UK BioBank we will
use the data excluding the UK BioBank to avoid overlap of the exposure and outcome populations.

First we will download the data for LDL cholesterol - **Warning** The GWAS data files are large and
will take some time to download.

```{r download_data, eval = FALSE}
# The URL for the GWAS summary statistics README document
url_rm <- "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/README.txt"

# Download README to the package downloads directory (subdirectory "glgc").
get_web_file(url_rm, directory="glgc")

# The URLs for the GWAS summary statistics data
lipids_gwas_urls <- list(
  ldl    = "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz",
  hdl    = "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_HDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz",
  tc     = "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_TC_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz",
  nonhdl = "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_nonHDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz",
  tg     = "http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_logTG_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"
  )

# Download GLGC GWAS data to package downloads directory; return the file paths
# If the file has already been downloaded the existing file path will be returned
lipids_gwas_fps <- purrr::map(lipids_gwas_urls, get_web_file, directory="glgc")
```

There are more than 47 million SNPs in the datasets. We now filter the GWAS data SNPs associated
with circulating lipid levels at a statistical significance of `p<5e-8`. Reading this file can take
a long time and so we can cache (write out) the result to a .csv file within the package structure
so that if we need to run things again it is much quicker.

```{r significant_snps, eval = FALSE}
# set significance level for filtering SNPs 
p <- 5e-8

# Read the .gz data files and filter by significance level
sig_snps <- purrr::map2(.x = lipids_gwas_fps,                   
                        .y = names(lipids_gwas_fps),  
                        .f = ~extract_significant_snps(.x, .y, pval=p, pval_col="pvalue", snp_col="rsID"))

# View structure
str(sig_snps[[1]])
```

If we inspect the column names we see that they need some adjustment to ensure they are compatible
with the TwoSampleMR package naming conventions. `TwoSampleMR` provides a function to appropriately
rename the columns: `TwoSampleMR::format_data`.
```{r standardise, eval = FALSE}
# The current column names
colnames(sig_snps$ldl)

# Standardise exposure data column names for use with TwoSampleMR package
exposure_data <- purrr::map(.x = sig_snps,
                            .f = ~ TwoSampleMR::format_data(
                                    dat               = .x,
                                    type              = "exposure",
                                    snp_col           = "rsID",
                                    beta_col          = "EFFECT_SIZE",
                                    se_col            = "SE",
                                    eaf_col           = "POOLED_ALT_AF",
                                    effect_allele_col = "ALT",
                                    other_allele_col  = "REF",
                                    pval_col          = "pvalue",
                                    samplesize_col    = "N",
                                    chr_col           = "CHROM",
                                    pos_col           = "POS_b37"))

# View structure
str(exposure_data[[1]])
```

## Cis-acting lipid metabolism gene SNPs
We will create a set of exposure SNPs that will be subset of the SNPs that associate with the 5
lipid components. This subset of SNPs will be in close proximity (number of bases) to the genes of
interest (HMG-CoA Reductase, NPC1L1, PCSK9). Data regarding the positions of the genes is available
[online](https://www.genecards.org/cgi-bin/carddisp.pl?gene=HMGCR). The latest lipid data from the
GLGC uses the Genome Reference Consortium Human Reference 37 (GRCh37) and so we must use this when
referring to the position of the genes. First I create a list of objects of class `Gene` - this is a
simple custom S4 class containing gene information. In addition, we will look at all of the SNPs
across the whole genome that associate with lipid profile components.

```{r cis_hmg, eval = FALSE}
# The HMG-CoA-R, NPC1L1, and PCSK9 genes; set the 'cis' base tolerance here.
genes_list = list(
  Gene(name="HMGCR",  chromosome=5, start=74632993, end=74657941, cis_tol=1e5),
  Gene(name="NPC1L1", chromosome=7, start=44552134, end=44580929, cis_tol=1e5),
  Gene(name="PCSK9",  chromosome=1, start=55505221, end=55530525, cis_tol=1e5),
  Gene(name="GenomeWide", genome_wide=TRUE)
)

# Create a new dataset of SNPs close to the genes
cis_exp_dat <- create_cis_gene_exposure(exposure_data, genes_list)
```

## Clumping SNPs
Many of the SNPs will be highly correlated, i.e. be genetically linked and provide no additional
information over and above the whole set of correlated SNPs. To reduce the number of SNPs to only
those most significantly associated with lipid levels we use Linkage Disequilibrium clumping.
[Clumping](https://kp4cd.org/sites/default/files/documents/LD_clumping.pdf) reports the most
significant genetic associations in a region in terms of a smaller number of “clumps” of genetically
linked SNPs.

SNPs that are correlated with an R^2^ greater than some value, `clump_r2`, will be clumped - i.e.
the SNP with the lower p-value will be removed. The process is iterative and has several parameters:
`clump_kb`, `clump_r2`, `clump_p`. We will use the defaults, except for `clump_r2` with the, which
we will set more leniently, instead of the default of 0.001, due to the fact we are looking at
cis-acting SNPs which will be more correlated by virtue of their proximity to the genes of interest.

We could do this on the Bristol IEU severs using the function `TwoSampleMR::clump_data`. 
```{r clumping_1, eval = FALSE}
# Set the clumping parameters for the different scenarios
clump_args <- data.frame("exposure" = names(cis_exp_dat)) |>
  dplyr::mutate(clump_r2 = dplyr::case_when(grepl("HMGCR",       exposure) ~ 0.2, 
                                            grepl("NPC1L1",      exposure) ~ 0.2,
                                            grepl("PCSK9",       exposure) ~ 0.2,
                                            grepl("GenomeWide",  exposure) ~ 0.001), 
                clump_kb = 10000, 
                clump_p1 = 1, 
                clump_p2 = 1, 
                pop      = "EUR")

# View the proposed arguments
clump_args

# Run clumping
clumped_exp_dat <- clump_snps(cis_exp_dat, clump_r2=clump_args$clump_r2)
```

<br>

# Outcome data
## Sepsis
The outcome we are interested in is sepsis. There are several GWAS relating whether a patient
develops sepsis to their underlying SNPs. Several of these studies are available on the [IEU GWAS
database](https://gwas.mrcieu.ac.uk) which can be queried using either the `TwoSampleMR` or
`ieugwasr` R packages.
```{r outcome, eval = FALSE}
# List of available GWAS summary statistics
available_gwas <- ieugwasr::gwasinfo() |>
  
  # Filter for those that mention 'sepsis' in the 'trait' description column
  dplyr::filter(grepl("sepsis|pneumonia", .data$trait, ignore.case=TRUE))

# We will pick a few sepsis GWAS
outcome_gwas_id <- c("ieu-b-5086", "ieu-b-5088", "ieu-b-4982", 
                     "ieu-b-4980", "ieu-b-4981", "ieu-b-69", 
                     "finn-b-SEPTICHE_PNEUMONIA")
```
## Extracting outcome data
Having identified the GWAS summary statistics we next need to extract the data for our SNPs of
interest (those that are highly predictive of the exposure, in this case the components of the blood
lipid profile).
```{r outcome_extracting, eval = FALSE}
# Extract data for cis-gene SNPs
outcome_data <- purrr::map(.x = clumped_exp_dat,
                           .f = ~ TwoSampleMR::extract_outcome_data(
                              snps            = .x$SNP,
                              outcomes        = outcome_gwas_id,
                              proxies         = TRUE,
                              rsq             = 0.8,
                              align_alleles   = 1,
                              palindromes     = 1,
                              maf_threshold   = 0.3,
                              access_token    = ieugwasr::check_access_token(),
                              splitsize       = 10000,
                              proxy_splitsize = 500))
```

<br>

# Harmonising
We have the effects of the (significant) SNPs on lipid levels and the effect of these same SNPs on
the outcome sepsis. However, it is important to harmonise the effects. This means that the effect of
a SNP on the exposure and the effect of that SNP on the outcome must each correspond to the same
allele (i.e. the A&Ts and C&Gs cannot be the wrong way round). *Note* The IEU GWAS database contains
data that is already harmonised. For more information on
[harmonising](https://mrcieu.github.io/TwoSampleMR/articles/harmonise.html).
```{r harmonise, eval = FALSE}
# Filter out the Gene:Lipid SNP associations with no significant SNPs
zero_snps <- purrr::map_lgl(clumped_exp_dat, ~ nrow(.x)==0)

# Harmonise the data for the LDL SNPs across the whole genome
harm_dat <- purrr::map2(.x = clumped_exp_dat[!zero_snps],
                        .y = outcome_data[!zero_snps],
                        .f = ~ TwoSampleMR::harmonise_data(.x, .y))
```   

<br>

# Two Sample Mendelian Randomisation
```{r mr, eval = FALSE}
# The MR result for all of the different scenarios
res_dat <- purrr::map(.x = harm_dat, 
                      .f = ~ TwoSampleMR::mr(.x))
```

## Effects and significance
```{r weighted_effects, eval = FALSE}
# Prepare the data
plot_dat <- purrr::map_df(res_dat, ~ .x |> 
                            dplyr::select(exposure, outcome, method, b, pval) |>
                            # Create two columns: gene
                            tidyr::separate(exposure, into=c("gene", "lipid"), sep="_") |>
                            dplyr::mutate(dplyr::across(c("gene", "lipid", "outcome"), ~as.factor(.x)))) |>
  # Ensure present in the data, even if NA
  tidyr::complete(gene, lipid, outcome, method) |>
  # Just look at IVW for now
  dplyr::filter(method == "Inverse variance weighted", ) |>
  # Deselect to make table nicer
  dplyr::select(-method)
```

## Interpretation
Genetically proxied inhibition of important lipid metabolism drug targets has no consistent effect
on the risk of developing sepsis across multiple cohorts.  

To be completed...

<br>

# Plots
## Heatmap
```{r plot_heatmap, include=TRUE, echo=TRUE, fig.height=35, fig.width=7, eval = FALSE}
library(ggplot2)

# Set the significance cutoff
p_val_cutoff = 0.05

# Plot heat map of lipid component against lipid regulation genes
p <- ggplot(plot_dat, aes(x=lipid, y=gene)) +
    # Set the fill tile colour to the p-value
    geom_tile(aes(fill=pval)) + 
    # Set the text to the beta value
    geom_text(aes(label = paste("\u03B2", round(b, 3)))) +
    # Set the text to the beta value
    geom_text(aes(label = paste("p =", round(pval, 3))), nudge_y=-0.15, size=3.5, fontface = "italic") +
    # Ensure a very sharp colour transition at <0.05 --> green
    # Bottom 5% colour scale is green; top 95% colour scale is viridis gradient
    scale_fill_gradientn(colors = c(rep("green",50),
                                    scales::viridis_pal(option = "mako", begin=1, end=0.2)(950)),
                         values = seq(0,1, length.out=1000),
                         # Set limits to ensure that default min(data)/max(data) is not used
                         limits = c(0, 1), 
                         breaks = c(1.0, 0.75, 0.50, 0.25, p_val_cutoff),
                         labels = scales::label_number(accuracy = 0.01)) + 
    labs(title    ="MR estimate by lipid metabolism gene and lipid profile component", 
         subtitle = "Colour represents p-value, with bright green colour at cut-off p<0.05",
         x        ="Lipid component", 
         y        = "Gene") +
  facet_wrap(~outcome, ncol = 1, scales='free')
```
```{r write_heatmap, include=FALSE, eval = FALSE}
# Write out the figure (size )
png(file="figures/mr_heatmap.png", width=7, height=35, units="in", res=100)
p
dev.off()
```

```{r show_heatmap, fig.align="center", eval=TRUE, echo=FALSE, eval = TRUE}
# Show figure
knitr::include_graphics("figures/mr_heatmap.png")
```

<br>

## MR scatter plots

to be added
```{r plot_mr_scatter, include=FALSE, echo=TRUE, eval = FALSE}
# Plots the MR scatter plots
scatter_plots <- purrr::map2(.x = res_dat, 
                             .y = harm_dat,
                             .f = ~ TwoSampleMR::mr_scatter_plot(.x, .y))
```

```{r write_mr_scatter, include=FALSE, echo=FALSE, eval = FALSE}
# Unlist plots
scatter_plots <- unlist(scatter_plots, recursive=FALSE)

# Write out the figures
# for(i in 1:3)
# {
#     get(scatter_plots[[i]] 
#     ggplot2::ggsave(paste0("figures/mr_scatter_plot_", i, ".png"), width=7, height=5, units="in", dpi=100)
# }
```

```{r show_mr_scatter, fig.align="center", eval=TRUE, echo=FALSE, results='asis', eval = FALSE}
# Get the file paths
# files <- list.files(path="figures", pattern="mr_scatter_plot_\\d+.png", full.names=TRUE)
# 
# # Show figure - n.b. knitr::include_graphics doesn't work in a loop so need to do manually (https://stackoverflow.com/questions/51268623/insert-images-using-knitrinclude-graphics-in-a-for-loop)
# cat(paste0("![](", files, ")"), "\n")
```
