<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MRLipidInfection">
<title>Genetically Proxied Lipid Therapy And Sepsis • MRLipidInfection</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetically Proxied Lipid Therapy And Sepsis">
<meta property="og:description" content="MRLipidInfection">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MRLipidInfection</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/lipid-therapy-sepsis-mr.html">Genetically Proxied Lipid Therapy And Sepsis</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Genetically Proxied Lipid Therapy And Sepsis</h1>
                        <h4 data-toc-skip class="author">Nicholas
Sunderland</h4>
            
      
      
      <div class="d-none name"><code>lipid-therapy-sepsis-mr.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the latest version of MRLipidInfection in R
using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"mrcieu/gwasvcf"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>First we need to load the MRLipidInfection package and setup package
requirements by calling ‘setupMRLipidInfection()’. At the first
installation this will take some time as this will download the 1000
genomes LD reference panel into the package file structure and ensure
that PLINK is installed locally.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nicksunderland.github.io/MRLipidInfection/">MRLipidInfection</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exposure-data">Exposure data<a class="anchor" aria-label="anchor" href="#exposure-data"></a>
</h2>
<p>Next we must obtain the data for the exposure. We are interested
whether exposure to lipid lowering medications influences risk of
developing sepsis. This data is not readily available on a population
level and has many confounding factors. We will therefore estimate the
effect of genetically proxied lipid medication therapy by looking for
Single Nucleotide Polymorphisms (SNPs) in close proximity to known lipid
metabolism genes (e.g. HMG-CoA reductase) that are significantly
associated with circulating lipid levels. We will use the the latest
Global Lipids Genetics Consortium <a href="http://csg.sph.umich.edu/willer/public/glgc-lipids2021/" class="external-link">Ancestry-specific
GWAS summary statistics</a> for HDL-C, LDL-C, nonHDL-C, TC and TG.</p>
<p>First we will download the data for LDL cholesterol -
<strong>Warning</strong> GWAS data file is large and will take some time
to download.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The URL for the GWAS summary statistics README document</span></span>
<span><span class="va">url_1</span> <span class="op">&lt;-</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/README.txt"</span></span>
<span></span>
<span><span class="co"># Download README to the package downloads directory (subdirectory "glgc").</span></span>
<span><span class="fu"><a href="../reference/get_web_file.html">get_web_file</a></span><span class="op">(</span><span class="va">url_1</span>, directory<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The URL for the GWAS summary statistics data</span></span>
<span><span class="va">url_2</span> <span class="op">&lt;-</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span></span>
<span></span>
<span><span class="co"># Download '.gz' file to the package downloads directory (subdirectory "glgc").</span></span>
<span><span class="va">fp_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_web_file.html">get_web_file</a></span><span class="op">(</span><span class="va">url_2</span>, directory<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># GWAS data file now at:</span></span>
<span><span class="va">fp_2</span></span></code></pre></div>
<p>There are more than 47 million SNPs in the dataset. We now filter the
GWAS data SNPs associated with circulating LDL cholesterol levels at a
statisticsal significance of &lt;5e-8. Reading this file can take a long
time and so we can write out the result to a .csv file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read the .gz data file</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">fp_2</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span></span>
<span>  <span class="co"># Ensure that the pvalue is numeric </span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>pvalue <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span></span>
<span>  <span class="co"># Filter for SNPs with a pvalue &lt;5e-8</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">5e-8</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="co"># Ensure unique SNPs</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">rsID</span>, .keep_all<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The path to write to (place it alongside the main data file in the downloads directory)</span></span>
<span><span class="va">fp_d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_downloads_dir.html">get_downloads_dir</a></span><span class="op">(</span>subdir<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span>, <span class="st">"/glgc_sig_snps.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Write out the significant SNPs</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu">write_csv</span><span class="op">(</span><span class="va">d</span>, <span class="va">fp_d</span><span class="op">)</span></span></code></pre></div>
<p>In the case we want to re-run the analysis at any point we can just
read back in the smaller .csv file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The path to the significant exposure SNPs</span></span>
<span><span class="va">fp_d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="../reference/get_downloads_dir.html">get_downloads_dir</a></span><span class="op">(</span>subdir<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span>, <span class="st">"/glgc_sig_snps.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Read back in</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu">read_csv</span><span class="op">(</span><span class="va">fp_d</span>, show_col_types<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>If we inspect the column names we see that they need some adjustment
to ensure they are compatible with the TwoSampleMR package naming
conventions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The current column names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Standardise the exposure data column names (set clump to FALSE as we will do this later)</span></span>
<span><span class="va">exposure_data</span> <span class="op">&lt;-</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">read_exposure_data</span><span class="op">(</span></span>
<span>    filename          <span class="op">=</span> <span class="va">fp_d</span>,</span>
<span>    clump             <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    sep               <span class="op">=</span> <span class="st">","</span>,</span>
<span>    snp_col           <span class="op">=</span> <span class="st">"rsID"</span>,</span>
<span>    beta_col          <span class="op">=</span> <span class="st">"EFFECT_SIZE"</span>,</span>
<span>    se_col            <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>    eaf_col           <span class="op">=</span> <span class="st">"POOLED_ALT_AF"</span>,</span>
<span>    effect_allele_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>    other_allele_col  <span class="op">=</span> <span class="st">"REF"</span>,</span>
<span>    pval_col          <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>    samplesize_col    <span class="op">=</span> <span class="st">"N"</span>,</span>
<span>    chr_col           <span class="op">=</span> <span class="st">"CHROM"</span>,</span>
<span>    pos_col           <span class="op">=</span> <span class="st">"POS_b37"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add 'LDL' as the exposure tag</span></span>
<span><span class="va">exposure_data</span><span class="op">$</span><span class="va">exposure</span> <span class="op">&lt;-</span> <span class="st">"LDL"</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicholas Sunderland.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
