<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MRLipidInfection">
<title>Genetically Proxied Lipid Therapy And Sepsis • MRLipidInfection</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetically Proxied Lipid Therapy And Sepsis">
<meta property="og:description" content="MRLipidInfection">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MRLipidInfection</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/lipid-therapy-sepsis-mr.html">Genetically Proxied Lipid Therapy And Sepsis</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Genetically Proxied Lipid Therapy And Sepsis</h1>
                        <h4 data-toc-skip class="author">Nicholas
Sunderland</h4>
            
      
      
      <div class="d-none name"><code>lipid-therapy-sepsis-mr.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>You can install the latest version of MRLipidInfection in R
using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"nicksunderland/MRLipidInfection"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>First we need to load the MRLipidInfection package and setup package
requirements by calling <code><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection()</a></code>. At the
first installation this will take some time as this will download the <a href="http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz" class="external-link">1000 genomes LD
reference panel</a> into the package file structure and ensure that <a href="https://www.cog-genomics.org/plink/" class="external-link">PLINK</a> is installed
locally.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nicksunderland.github.io/MRLipidInfection/">MRLipidInfection</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level2">
<h2 id="exposure-data">Exposure data<a class="anchor" aria-label="anchor" href="#exposure-data"></a>
</h2>
<div class="section level3">
<h3 id="lipid-profile-data">Lipid profile data<a class="anchor" aria-label="anchor" href="#lipid-profile-data"></a>
</h3>
<p>Next we must obtain the data for the exposure. We are interested
whether exposure to lipid lowering medications influences risk of
developing sepsis. This data is not readily available on a population
level and has many confounding factors. We will therefore estimate the
effect of genetically proxied lipid medication therapy by looking for
Single Nucleotide Polymorphisms (SNPs) in close proximity to known lipid
metabolism genes (e.g. HMG-CoA reductase) that are significantly
associated with circulating lipid levels. We will use the the latest
Global Lipids Genetics Consortium <a href="http://csg.sph.umich.edu/willer/public/glgc-lipids2021/" class="external-link">Ancestry-specific
GWAS summary statistics</a> for HDL-C, LDL-C, nonHDL-C, TC and TG.</p>
<p>First we will download the data for LDL cholesterol -
<strong>Warning</strong> GWAS data file is large and will take some time
to download.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The URL for the GWAS summary statistics README document</span></span>
<span><span class="va">url_rm</span> <span class="op">&lt;-</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/README.txt"</span></span>
<span></span>
<span><span class="co"># Download README to the package downloads directory (subdirectory "glgc").</span></span>
<span><span class="fu"><a href="../reference/get_web_file.html">get_web_file</a></span><span class="op">(</span><span class="va">url_rm</span>, directory<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The URLs for the GWAS summary statistics data</span></span>
<span><span class="va">lipids_gwas_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  ldl    <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  hdl    <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/HDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  tc     <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/TC_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  nonhdl <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/nonHDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  tg     <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/logTG_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download GLGC GWAS data to package downloads directory; return the file paths</span></span>
<span><span class="va">lipids_gwas_fps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lipids_gwas_urls</span>, <span class="va">get_web_file</span>, directory<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span></span>
<span><span class="va">lipids_gwas_fps</span></span></code></pre></div>
<p>There are more than 47 million SNPs in the dataset. We now filter the
GWAS data SNPs associated with circulating LDL cholesterol levels at a
statisticsal significance of <code>p&lt;5e-8</code>. Reading this file
can take a long time and so we can write out the result to a .csv
file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set significance level for filtering SNPs </span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span>
<span></span>
<span><span class="co"># Read the .gz data files and filter by significance level</span></span>
<span><span class="va">sig_snps</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">lipids_gwas_fps</span>,                   </span>
<span>                        .y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lipids_gwas_fps</span><span class="op">)</span>,           </span>
<span>                        .f <span class="op">=</span> <span class="op">~</span><span class="fu"><a href="../reference/extract_significant_snps.html">extract_significant_snps</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span>, pval<span class="op">=</span><span class="va">p</span>, pval_col<span class="op">=</span><span class="st">"pvalue"</span>, snp_col<span class="op">=</span><span class="st">"rsID"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the name </span></span>
<span><span class="va">sig_snps</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">sig_snps</span>, </span>
<span>                        .y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sig_snps</span><span class="op">)</span>,</span>
<span>                        .f <span class="op">=</span> <span class="op">~</span> <span class="va">.x</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>phenotype <span class="op">=</span> <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View one of the tables:</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sig_snps</span><span class="op">$</span><span class="va">ldl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The file path to the data in .csv format</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">sig_snps</span><span class="op">$</span><span class="va">ldl</span>, <span class="st">"filepath"</span><span class="op">)</span></span></code></pre></div>
<p>If we inspect the column names we see that they need some adjustment
to ensure they are compatible with the TwoSampleMR package naming
conventions.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The current column names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sig_snps</span><span class="op">$</span><span class="va">ldl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Standardise exposure data column names</span></span>
<span><span class="va">exposure_data</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">sig_snps</span>,</span>
<span>                            .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">format_data</span><span class="op">(</span></span>
<span>                                    dat               <span class="op">=</span> <span class="va">.x</span>,</span>
<span>                                    type              <span class="op">=</span> <span class="st">"exposure"</span>,</span>
<span>                                    phenotype_col     <span class="op">=</span> <span class="st">"phenotype"</span>,</span>
<span>                                    snp_col           <span class="op">=</span> <span class="st">"rsID"</span>,</span>
<span>                                    beta_col          <span class="op">=</span> <span class="st">"EFFECT_SIZE"</span>,</span>
<span>                                    se_col            <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>                                    eaf_col           <span class="op">=</span> <span class="st">"POOLED_ALT_AF"</span>,</span>
<span>                                    effect_allele_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>                                    other_allele_col  <span class="op">=</span> <span class="st">"REF"</span>,</span>
<span>                                    pval_col          <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>                                    samplesize_col    <span class="op">=</span> <span class="st">"N"</span>,</span>
<span>                                    chr_col           <span class="op">=</span> <span class="st">"CHROM"</span>,</span>
<span>                                    pos_col           <span class="op">=</span> <span class="st">"POS_b37"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cis-acting-hmg-coa-reductase-snps">Cis-acting HMG-CoA Reductase SNPs<a class="anchor" aria-label="anchor" href="#cis-acting-hmg-coa-reductase-snps"></a>
</h3>
<p>We will create a second set of exposure SNPs that will be subset of
the SNPs that associate with LDL cholesterol levels. This subset of SNPs
will be within a certain proximity (number of bases) to the HMG-CoA
Reductase gene. Data regarding the position of the HMG-CoA-R gene is
freely available <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=HMGCR" class="external-link">online</a>.
The Lipid data from the GLGC uses the Genome Reference Consortium Human
Reference 37 (GRCh37) and so we must use this when referring to the
position of the HMG-CoA-R gene.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The HMG-CoA-R gene is on chromosome 5, starts at base 746,32,993 and ends at base 74,657,941.</span></span>
<span><span class="va">chrom_hmg</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">start_hmg</span> <span class="op">&lt;-</span> <span class="fl">74632993</span></span>
<span><span class="va">end_hmg</span>   <span class="op">&lt;-</span> <span class="fl">74657941</span></span>
<span></span>
<span><span class="co"># We want to capture SNPs near the HMG-CoA gene and so we must set a tolerance window either side in</span></span>
<span><span class="co"># which to also search</span></span>
<span><span class="co">## changed for testing, normal 3e5</span></span>
<span><span class="va">tol</span> <span class="op">&lt;-</span> <span class="fl">3e6</span></span>
<span></span>
<span><span class="co"># Create a new dataset of SNPs close to the HMG-CoA gene</span></span>
<span><span class="va">cis_exp_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>  .x <span class="op">=</span> <span class="va">exposure_data</span>,</span>
<span>  .f <span class="op">=</span> <span class="op">~</span> <span class="va">.x</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>id.exposure <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">chr.exposure</span> <span class="op">==</span> <span class="va">chrom_hmg</span> <span class="op">&amp;</span></span>
<span>                                               <span class="va">.data</span><span class="op">$</span><span class="va">pos.exposure</span> <span class="op">&gt;</span> <span class="op">(</span><span class="va">start_hmg</span> <span class="op">-</span> <span class="va">tol</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>                                               <span class="va">.data</span><span class="op">$</span><span class="va">pos.exposure</span> <span class="op">&lt;</span> <span class="op">(</span><span class="va">end_hmg</span>   <span class="op">+</span> <span class="va">tol</span><span class="op">)</span>, </span>
<span>                                               <span class="st">"cis_hmg"</span>, <span class="cn">NA_character_</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">id.exposure</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clumping-snps">Clumping SNPs<a class="anchor" aria-label="anchor" href="#clumping-snps"></a>
</h3>
<p>Many of the SNPs will be highly correlated, i.e. be genetically
linked and provide no additional information over and above the whole
set of correlated SNPs. To reduce the number of SNPs to only those most
significantly associated with lipid levels we use Linkage Disequilibrium
clumping. <a href="https://kp4cd.org/sites/default/files/documents/LD_clumping.pdf" class="external-link">Clumping</a>
reports the most significant genetic associations in a region in terms
of a smaller number of “clumps” of genetically linked SNPs.</p>
<p>SNPs that are correlated with an R<sup>2</sup> greater than some
value <code>clump_r2</code> will be clumped - i.e. the SNP with the
lower p-value will be removed. The process is iterative and has several
parameters: <code>clump_kb</code>, <code>clump_r2</code>,
<code>clump_p</code>. We will use the defaults, except for
<code>clump_r2</code> with the cis-HMG-CoA-R SNPs, which we will set to
0.1 instead of the default of 0.001.</p>
<p>We could do this on the Bristol IEU severs using the function
<code>TwoSampleMR::clump_data</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Only significant SNPs in close proximity to the HMG-CoA reductase gene</span></span>
<span><span class="co">#TODO: write function to deal with no results from clumping (at the minute the cis window is big)</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">clumped_exp_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">cis_exp_dat</span><span class="op">)</span>, </span>
<span>                              .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">clump_data</span><span class="op">(</span></span>
<span>                                dat      <span class="op">=</span> <span class="va">.x</span>,</span>
<span>                                clump_r2 <span class="op">=</span> <span class="va">clump_r2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clumped_exp_dat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cis_exp_dat</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, we can do this locally using the 1000 Genomes
Reference Panel and PLINK binary that we downloaded earlier.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">clumped_exp_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">cis_exp_dat</span><span class="op">)</span>,</span>
<span>                              .f <span class="op">=</span> <span class="op">~</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu">ld_clump</span><span class="op">(</span></span>
<span>                                dat       <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>rsid<span class="op">=</span><span class="va">.x</span><span class="op">$</span><span class="va">SNP</span>,</span>
<span>                                                          pval<span class="op">=</span><span class="va">.x</span><span class="op">$</span><span class="va">pval.exposure</span><span class="op">)</span>,</span>
<span>                                plink_bin <span class="op">=</span> <span class="fu">genetics.binaRies</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/genetics.binaRies/man/get_plink_binary.html" class="external-link">get_plink_binary</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                bfile     <span class="op">=</span> <span class="fu"><a href="../reference/get_b_file.html">get_b_file</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>                                clump_r2  <span class="op">=</span> <span class="va">clump_r2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clumped_exp_dat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cis_exp_dat</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="outcome-data">Outcome data<a class="anchor" aria-label="anchor" href="#outcome-data"></a>
</h2>
<div class="section level3">
<h3 id="sepsis">Sepsis<a class="anchor" aria-label="anchor" href="#sepsis"></a>
</h3>
<p>The outcome we are interested in is sepsis. There are several GWAS
relating whether a patient develops sepsis to their underlying SNPs.
Several of these studies are available on the <a href="https://gwas.mrcieu.ac.uk" class="external-link">IEU GWAS database</a> which can be
queried using either the <code>TwoSampleMR</code> or
<code>ieugwasr</code> R packages.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List of available GWAS summary statistics</span></span>
<span><span class="va">available_gwas</span> <span class="op">&lt;-</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu">gwasinfo</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="co"># Filter for those that mention 'sepsis' in the 'trait' description column</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"sepsis|pneumonia"</span>, <span class="va">.data</span><span class="op">$</span><span class="va">trait</span>, ignore.case<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We will pick one</span></span>
<span><span class="va">outcome_gwas_id</span> <span class="op">&lt;-</span> <span class="st">"ieu-b-4980"</span></span>
<span></span>
<span><span class="co"># View</span></span>
<span><span class="va">available_gwas</span><span class="op">[</span><span class="va">available_gwas</span><span class="op">$</span><span class="va">id</span> <span class="op">==</span> <span class="va">outcome_gwas_id</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extracting-outcome-data">Extracting outcome data<a class="anchor" aria-label="anchor" href="#extracting-outcome-data"></a>
</h3>
<p>Having identified the GWAS summary statistics we next need to extract
the data for our SNPs of interest (those that are highly predictive of
the exposure, in this case LDL cholesterol).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The cisHMG-CoA SNPs won't necessarily all be within the group of SNPs for the whole genome.</span></span>
<span><span class="co">##all(clumped_hmg_exp$SNP %in% clumped_exp$SNP)</span></span>
<span></span>
<span><span class="co"># Extract data for cis-HMG-CoA SNPs</span></span>
<span><span class="va">outcome_data</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">clumped_exp_dat</span>,</span>
<span>                           .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">extract_outcome_data</span><span class="op">(</span></span>
<span>                              snps            <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">SNP</span>,</span>
<span>                              outcomes        <span class="op">=</span> <span class="va">outcome_gwas_id</span>,</span>
<span>                              proxies         <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                              rsq             <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                              align_alleles   <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                              palindromes     <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                              maf_threshold   <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>                              access_token    <span class="op">=</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu">check_access_token</span><span class="op">(</span><span class="op">)</span>,</span>
<span>                              splitsize       <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>                              proxy_splitsize <span class="op">=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">outcome_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clumped_exp_dat</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="obtaining-outcome-data-not-on-the-ieu-gwas-database">Obtaining outcome data <em>not</em> on the IEU GWAS database<a class="anchor" aria-label="anchor" href="#obtaining-outcome-data-not-on-the-ieu-gwas-database"></a>
</h3>
<p>How to extract outcome data from other sources and find proxy
SNPs</p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="harmonising">Harmonising<a class="anchor" aria-label="anchor" href="#harmonising"></a>
</h2>
<p>We have the effects of the (significant) SNPs on LDL levels and the
effect of these same SNPs on the outcome sepsis. However, it is
important to harmonise the effects. This means that the effect of a SNP
on the exposure and the effect of that SNP on the outcome must each
correspond to the same allele (i.e. the A&amp;Ts and C&amp;Gs cannot be
the wrong way round). <em>Note</em> The IEU GWAS database contains data
that is already harmonised. For more information on <a href="https://mrcieu.github.io/TwoSampleMR/articles/harmonise.html" class="external-link">harmonising</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Harmonise the data for the LDL SNPs across the whole genome</span></span>
<span><span class="va">harm_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">clumped_exp_dat</span>,</span>
<span>                        .y <span class="op">=</span> <span class="va">outcome_data</span>,</span>
<span>                        .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">harmonise_data</span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level2">
<h2 id="two-sample-mendelian-randomisation">Two Sample Mendelian Randomisation<a class="anchor" aria-label="anchor" href="#two-sample-mendelian-randomisation"></a>
</h2>
<p>The whole point is to assess the weight influence of the SNPs on the
outcome</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The MR result using SNPs from across the genome</span></span>
<span><span class="va">res_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">harm_dat</span>, </span>
<span>                      .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">mr</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="plotting-the-result">Plotting the result<a class="anchor" aria-label="anchor" href="#plotting-the-result"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">res_dat</span>, </span>
<span>                     .y <span class="op">=</span> <span class="va">harm_dat</span>,</span>
<span>                     .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu">mr_scatter_plot</span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">plots</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="to-be-continued">To be continued<a class="anchor" aria-label="anchor" href="#to-be-continued"></a>
</h3>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicholas Sunderland.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
