<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MRLipidInfection">
<title>Genetically Proxied Lipid Therapy And Sepsis • MRLipidInfection</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetically Proxied Lipid Therapy And Sepsis">
<meta property="og:description" content="MRLipidInfection">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MRLipidInfection</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/MRLipidInfection.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Genetically Proxied Lipid Therapy And Sepsis</h1>
            
      
      
      <div class="d-none name"><code>MRLipidInfection.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>You can install the latest version of MRLipidInfection in R
using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"nicksunderland/MRLipidInfection"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>First we need to load the MRLipidInfection package and setup package
requirements by calling <code><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection()</a></code>. At the
first installation this will take some time as this will download the <a href="http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz" class="external-link">1000 genomes LD
reference panel</a> into the package file structure and ensure that <a href="https://www.cog-genomics.org/plink/" class="external-link">PLINK</a> is installed
locally. If you plan to use only the IEU servers for LD processing you
can skip these downloads.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nicksunderland.github.io/MRLipidInfection/">MRLipidInfection</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection</a></span><span class="op">(</span>skip_local_download<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level2">
<h2 id="exposure-data">Exposure data<a class="anchor" aria-label="anchor" href="#exposure-data"></a>
</h2>
<div class="section level3">
<h3 id="lipid-profile-data">Lipid profile data<a class="anchor" aria-label="anchor" href="#lipid-profile-data"></a>
</h3>
<p>Next we must obtain the data for the exposure. We are interested
whether exposure to lipid lowering medications influences risk of
developing sepsis. This data is not readily available on a population
level and has many confounding factors. We will therefore estimate the
effect of genetically proxied lipid medication therapy by looking for
Single Nucleotide Polymorphisms (SNPs) in close proximity to known lipid
metabolism genes (HMG-CoA Reductase, NPC1L1, PCSK9) that are
significantly associated with circulating lipid levels. We will use the
the latest Global Lipids Genetics Consortium <a href="http://csg.sph.umich.edu/willer/public/glgc-lipids2021/" class="external-link">Ancestry-specific
GWAS summary statistics</a> for HDL-C, LDL-C, nonHDL-C, TC and TG. As we
will be using outcome data from the UK BioBank we will use the data
excluding the UK BioBank to avoid overlap of the exposure and outcome
populations.</p>
<p>First we will download the data for LDL cholesterol -
<strong>Warning</strong> The GWAS data files are large and will take
some time to download.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The URL for the GWAS summary statistics README document</span></span>
<span><span class="va">url_rm</span> <span class="op">&lt;-</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/README.txt"</span></span>
<span></span>
<span><span class="co"># Download README to the package downloads directory (subdirectory "glgc").</span></span>
<span><span class="fu"><a href="../reference/get_web_file.html">get_web_file</a></span><span class="op">(</span><span class="va">url_rm</span>, directory<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The URLs for the GWAS summary statistics data</span></span>
<span><span class="va">lipids_gwas_urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  ldl    <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  hdl    <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_HDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  tc     <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_TC_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  nonhdl <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_nonHDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  tg     <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_logTG_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Download GLGC GWAS data to package downloads directory; return the file paths</span></span>
<span><span class="co"># If the file has already been downloaded the existing file path will be returned</span></span>
<span><span class="va">lipids_gwas_fps</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">lipids_gwas_urls</span>, <span class="va">get_web_file</span>, directory<span class="op">=</span><span class="st">"glgc"</span><span class="op">)</span></span></code></pre></div>
<p>There are more than 47 million SNPs in the datasets. We now filter
the GWAS data SNPs associated with circulating lipid levels at a
statistical significance of <code>p&lt;5e-8</code>. Reading this file
can take a long time and so we can cache (write out) the result to a
.csv file within the package structure so that if we need to run things
again it is much quicker.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># set significance level for filtering SNPs </span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span>
<span></span>
<span><span class="co"># Read the .gz data files and filter by significance level</span></span>
<span><span class="va">sig_snps</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">lipids_gwas_fps</span>,                   </span>
<span>                        .y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">lipids_gwas_fps</span><span class="op">)</span>,  </span>
<span>                        .f <span class="op">=</span> <span class="op">~</span><span class="fu"><a href="../reference/extract_significant_snps.html">extract_significant_snps</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span>, pval<span class="op">=</span><span class="va">p</span>, pval_col<span class="op">=</span><span class="st">"pvalue"</span>, snp_col<span class="op">=</span><span class="st">"rsID"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">sig_snps</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p>If we inspect the column names we see that they need some adjustment
to ensure they are compatible with the TwoSampleMR package naming
conventions. <code>TwoSampleMR</code> provides a function to
appropriately rename the columns:
<code><a href="https://rdrr.io/pkg/TwoSampleMR/man/format_data.html" class="external-link">TwoSampleMR::format_data</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The current column names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sig_snps</span><span class="op">$</span><span class="va">ldl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Standardise exposure data column names for use with TwoSampleMR package</span></span>
<span><span class="va">exposure_data</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">sig_snps</span>,</span>
<span>                            .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/format_data.html" class="external-link">format_data</a></span><span class="op">(</span></span>
<span>                                    dat               <span class="op">=</span> <span class="va">.x</span>,</span>
<span>                                    type              <span class="op">=</span> <span class="st">"exposure"</span>,</span>
<span>                                    snp_col           <span class="op">=</span> <span class="st">"rsID"</span>,</span>
<span>                                    beta_col          <span class="op">=</span> <span class="st">"EFFECT_SIZE"</span>,</span>
<span>                                    se_col            <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>                                    eaf_col           <span class="op">=</span> <span class="st">"POOLED_ALT_AF"</span>,</span>
<span>                                    effect_allele_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>                                    other_allele_col  <span class="op">=</span> <span class="st">"REF"</span>,</span>
<span>                                    pval_col          <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>                                    samplesize_col    <span class="op">=</span> <span class="st">"N"</span>,</span>
<span>                                    chr_col           <span class="op">=</span> <span class="st">"CHROM"</span>,</span>
<span>                                    pos_col           <span class="op">=</span> <span class="st">"POS_b37"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">exposure_data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cis-acting-lipid-metabolism-gene-snps">Cis-acting lipid metabolism gene SNPs<a class="anchor" aria-label="anchor" href="#cis-acting-lipid-metabolism-gene-snps"></a>
</h3>
<p>We will create a set of exposure SNPs that will be subset of the SNPs
that associate with the 5 lipid components. This subset of SNPs will be
in close proximity (number of bases) to the genes of interest (HMG-CoA
Reductase, NPC1L1, PCSK9). Data regarding the positions of the genes is
available <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=HMGCR" class="external-link">online</a>.
The latest lipid data from the GLGC uses the Genome Reference Consortium
Human Reference 37 (GRCh37) and so we must use this when referring to
the position of the genes. First I create a list of objects of class
<code>Gene</code> - this is a simple custom S4 class containing gene
information. In addition, we will look at all of the SNPs across the
whole genome that associate with lipid profile components.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The HMG-CoA-R, NPC1L1, and PCSK9 genes; set the 'cis' base tolerance here.</span></span>
<span><span class="va">genes_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/Gene-class.html">Gene</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"HMGCR"</span>,  chromosome<span class="op">=</span><span class="fl">5</span>, start<span class="op">=</span><span class="fl">74632993</span>, end<span class="op">=</span><span class="fl">74657941</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/Gene-class.html">Gene</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"NPC1L1"</span>, chromosome<span class="op">=</span><span class="fl">7</span>, start<span class="op">=</span><span class="fl">44552134</span>, end<span class="op">=</span><span class="fl">44580929</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/Gene-class.html">Gene</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"PCSK9"</span>,  chromosome<span class="op">=</span><span class="fl">1</span>, start<span class="op">=</span><span class="fl">55505221</span>, end<span class="op">=</span><span class="fl">55530525</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/Gene-class.html">Gene</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"GenomeWide"</span>, genome_wide<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a new dataset of SNPs close to the genes</span></span>
<span><span class="va">cis_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_cis_gene_exposure.html">create_cis_gene_exposure</a></span><span class="op">(</span><span class="va">exposure_data</span>, <span class="va">genes_list</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clumping-snps">Clumping SNPs<a class="anchor" aria-label="anchor" href="#clumping-snps"></a>
</h3>
<p>Many of the SNPs will be highly correlated, i.e. be genetically
linked and provide no additional information over and above the whole
set of correlated SNPs. To reduce the number of SNPs to only those most
significantly associated with lipid levels we use Linkage Disequilibrium
clumping. <a href="https://kp4cd.org/sites/default/files/documents/LD_clumping.pdf" class="external-link">Clumping</a>
reports the most significant genetic associations in a region in terms
of a smaller number of “clumps” of genetically linked SNPs.</p>
<p>SNPs that are correlated with an R<sup>2</sup> greater than some
value, <code>clump_r2</code>, will be clumped - i.e. the SNP with the
lower p-value will be removed. The process is iterative and has several
parameters: <code>clump_kb</code>, <code>clump_r2</code>,
<code>clump_p</code>. We will use the defaults, except for
<code>clump_r2</code> with the, which we will set more leniently,
instead of the default of 0.001, due to the fact we are looking at
cis-acting SNPs which will be more correlated by virtue of their
proximity to the genes of interest.</p>
<p>We could do this on the Bristol IEU severs using the function
<code><a href="https://rdrr.io/pkg/TwoSampleMR/man/clump_data.html" class="external-link">TwoSampleMR::clump_data</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set the clumping parameters for the different scenarios</span></span>
<span><span class="va">clump_args</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">"exposure"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cis_exp_dat</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>clump_r2 <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"HMGCR"</span>,       <span class="va">exposure</span><span class="op">)</span> <span class="op">~</span> <span class="fl">0.2</span>, </span>
<span>                                            <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"NPC1L1"</span>,      <span class="va">exposure</span><span class="op">)</span> <span class="op">~</span> <span class="fl">0.2</span>,</span>
<span>                                            <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"PCSK9"</span>,       <span class="va">exposure</span><span class="op">)</span> <span class="op">~</span> <span class="fl">0.2</span>,</span>
<span>                                            <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"GenomeWide"</span>,  <span class="va">exposure</span><span class="op">)</span> <span class="op">~</span> <span class="fl">0.001</span><span class="op">)</span>, </span>
<span>                clump_kb <span class="op">=</span> <span class="fl">10000</span>, </span>
<span>                clump_p1 <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                clump_p2 <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                pop      <span class="op">=</span> <span class="st">"EUR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the proposed arguments</span></span>
<span><span class="va">clump_args</span></span>
<span></span>
<span><span class="co"># Run clumping</span></span>
<span><span class="va">clumped_exp_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clump_snps.html">clump_snps</a></span><span class="op">(</span><span class="va">cis_exp_dat</span>, clump_r2<span class="op">=</span><span class="va">clump_args</span><span class="op">$</span><span class="va">clump_r2</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="outcome-data">Outcome data<a class="anchor" aria-label="anchor" href="#outcome-data"></a>
</h2>
<div class="section level3">
<h3 id="sepsis">Sepsis<a class="anchor" aria-label="anchor" href="#sepsis"></a>
</h3>
<p>The outcome we are interested in is sepsis. There are several GWAS
relating whether a patient develops sepsis to their underlying SNPs.
Several of these studies are available on the <a href="https://gwas.mrcieu.ac.uk" class="external-link">IEU GWAS database</a> which can be
queried using either the <code>TwoSampleMR</code> or
<code>ieugwasr</code> R packages.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List of available GWAS summary statistics</span></span>
<span><span class="va">available_gwas</span> <span class="op">&lt;-</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ieugwasr/man/gwasinfo.html" class="external-link">gwasinfo</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="co"># Filter for those that mention 'sepsis' in the 'trait' description column</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"sepsis|pneumonia"</span>, <span class="va">.data</span><span class="op">$</span><span class="va">trait</span>, ignore.case<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We will pick a few sepsis GWAS</span></span>
<span><span class="va">outcome_gwas_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ieu-b-5086"</span>, <span class="st">"ieu-b-5088"</span>, <span class="st">"ieu-b-4982"</span>, </span>
<span>                     <span class="st">"ieu-b-4980"</span>, <span class="st">"ieu-b-4981"</span>, <span class="st">"ieu-b-69"</span>, </span>
<span>                     <span class="st">"finn-b-SEPTICHE_PNEUMONIA"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extracting-outcome-data">Extracting outcome data<a class="anchor" aria-label="anchor" href="#extracting-outcome-data"></a>
</h3>
<p>Having identified the GWAS summary statistics we next need to extract
the data for our SNPs of interest (those that are highly predictive of
the exposure, in this case the components of the blood lipid
profile).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract data for cis-gene SNPs</span></span>
<span><span class="va">outcome_data</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">clumped_exp_dat</span>,</span>
<span>                           .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html" class="external-link">extract_outcome_data</a></span><span class="op">(</span></span>
<span>                              snps            <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">SNP</span>,</span>
<span>                              outcomes        <span class="op">=</span> <span class="va">outcome_gwas_id</span>,</span>
<span>                              proxies         <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                              rsq             <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                              align_alleles   <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                              palindromes     <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                              maf_threshold   <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>                              access_token    <span class="op">=</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ieugwasr/man/check_access_token.html" class="external-link">check_access_token</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                              splitsize       <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>                              proxy_splitsize <span class="op">=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="harmonising">Harmonising<a class="anchor" aria-label="anchor" href="#harmonising"></a>
</h2>
<p>We have the effects of the (significant) SNPs on lipid levels and the
effect of these same SNPs on the outcome sepsis. However, it is
important to harmonise the effects. This means that the effect of a SNP
on the exposure and the effect of that SNP on the outcome must each
correspond to the same allele (i.e. the A&amp;Ts and C&amp;Gs cannot be
the wrong way round). <em>Note</em> The IEU GWAS database contains data
that is already harmonised. For more information on <a href="https://mrcieu.github.io/TwoSampleMR/articles/harmonise.html" class="external-link">harmonising</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out the Gene:Lipid SNP associations with no significant SNPs</span></span>
<span><span class="va">zero_snps</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_lgl</a></span><span class="op">(</span><span class="va">clumped_exp_dat</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Harmonise the data for the LDL SNPs across the whole genome</span></span>
<span><span class="va">harm_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">clumped_exp_dat</span><span class="op">[</span><span class="op">!</span><span class="va">zero_snps</span><span class="op">]</span>,</span>
<span>                        .y <span class="op">=</span> <span class="va">outcome_data</span><span class="op">[</span><span class="op">!</span><span class="va">zero_snps</span><span class="op">]</span>,</span>
<span>                        .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html" class="external-link">harmonise_data</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level2">
<h2 id="two-sample-mendelian-randomisation">Two Sample Mendelian Randomisation<a class="anchor" aria-label="anchor" href="#two-sample-mendelian-randomisation"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The MR result for all of the different scenarios</span></span>
<span><span class="va">res_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">harm_dat</span>, </span>
<span>                      .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/mr.html" class="external-link">mr</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="effects-and-significance">Effects and significance<a class="anchor" aria-label="anchor" href="#effects-and-significance"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare the data</span></span>
<span><span class="va">plot_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_df</a></span><span class="op">(</span><span class="va">res_dat</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">|&gt;</span> </span>
<span>                            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">exposure</span>, <span class="va">outcome</span>, <span class="va">method</span>, <span class="va">b</span>, <span class="va">pval</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                            <span class="co"># Create two columns: gene</span></span>
<span>                            <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">exposure</span>, into<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"lipid"</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">"_"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"lipid"</span>, <span class="st">"outcome"</span><span class="op">)</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Ensure present in the data, even if NA</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/complete.html" class="external-link">complete</a></span><span class="op">(</span><span class="va">gene</span>, <span class="va">lipid</span>, <span class="va">outcome</span>, <span class="va">method</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Just look at IVW for now</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">method</span> <span class="op">==</span> <span class="st">"Inverse variance weighted"</span>, <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Deselect to make table nicer</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">method</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="interpretation">Interpretation<a class="anchor" aria-label="anchor" href="#interpretation"></a>
</h3>
<p>Genetically proxied inhibition of important lipid metabolism drug
targets has no consistent effect on the risk of developing sepsis across
multiple cohort.</p>
<p>To be completed…</p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="plots">Plots<a class="anchor" aria-label="anchor" href="#plots"></a>
</h2>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the significance cutoff</span></span>
<span><span class="va">p_val_cutoff</span> <span class="op">=</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Plot heat map of lipid component against lipid regulation genes</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plot_dat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">lipid</span>, y<span class="op">=</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># Set the fill tile colour to the p-value</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">pval</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="co"># Set the text to the beta value</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"\u03B2"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">b</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># Set the text to the beta value</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"p ="</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">pval</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, nudge_y<span class="op">=</span><span class="op">-</span><span class="fl">0.15</span>, size<span class="op">=</span><span class="fl">3.5</span>, fontface <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># Ensure a very sharp colour transition at &lt;0.05 --&gt; green</span></span>
<span>    <span class="co"># Bottom 5% colour scale is green; top 95% colour scale is viridis gradient</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"green"</span>,<span class="fl">50</span><span class="op">)</span>,</span>
<span>                                    <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/viridis_pal.html" class="external-link">viridis_pal</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"mako"</span>, begin<span class="op">=</span><span class="fl">1</span>, end<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span><span class="op">(</span><span class="fl">950</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                         values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>, length.out<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>,</span>
<span>                         <span class="co"># Set limits to ensure that default min(data)/max(data) is not used</span></span>
<span>                         limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                         breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.0</span>, <span class="fl">0.75</span>, <span class="fl">0.50</span>, <span class="fl">0.25</span>, <span class="va">p_val_cutoff</span><span class="op">)</span>,</span>
<span>                         labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/label_number.html" class="external-link">label_number</a></span><span class="op">(</span>accuracy <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title    <span class="op">=</span><span class="st">"MR estimate by lipid metabolism gene and lipid profile component"</span>, </span>
<span>         subtitle <span class="op">=</span> <span class="st">"Colour represents p-value, with bright green colour at cut-off p&lt;0.05"</span>,</span>
<span>         x        <span class="op">=</span><span class="st">"Lipid component"</span>, </span>
<span>         y        <span class="op">=</span> <span class="st">"Gene"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">outcome</span>, ncol <span class="op">=</span> <span class="fl">1</span>, scales<span class="op">=</span><span class="st">'free'</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/mr_heatmap.png" width="350" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="mr-scatter-plots">MR scatter plots<a class="anchor" aria-label="anchor" href="#mr-scatter-plots"></a>
</h3>
<p>to be added</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicholas Sunderland.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
