<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MRLipidInfection">
<title>Genetically Proxied Lipid Therapy And Sepsis • MRLipidInfection</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetically Proxied Lipid Therapy And Sepsis">
<meta property="og:description" content="MRLipidInfection">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MRLipidInfection</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/MRLipidInfection.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Genetically Proxied Lipid Therapy And Sepsis</h1>
            
      
      
      <div class="d-none name"><code>MRLipidInfection.Rmd</code></div>
    </div>

    
    
<style>
body{text-align: justify}
</style>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>You can install the latest version of MRLipidInfection in R
using:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"nicksunderland/MRLipidInfection"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setup-mrlipidinfection-package">Setup MRLipidInfection package<a class="anchor" aria-label="anchor" href="#setup-mrlipidinfection-package"></a>
</h3>
<p>First we need to load the MRLipidInfection package and setup package
requirements by calling <code><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection()</a></code>. At the
first installation this will take some time as this will download the <a href="http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz" class="external-link">1000 genomes LD
reference panel</a> into the package file structure and ensure that <a href="https://www.cog-genomics.org/plink/" class="external-link">PLINK</a> is installed
locally. If you plan to use only the Bristol IEU servers for LD
processing you can skip these downloads.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nicksunderland.github.io/MRLipidInfection/">MRLipidInfection</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/setupMRLipidInfection.html">setupMRLipidInfection</a></span><span class="op">(</span>skip_local_download<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="configuration-file">Configuration file<a class="anchor" aria-label="anchor" href="#configuration-file"></a>
</h3>
<p>The column names in any given GWAS file may not be consistent with
the <code>TwoSampleMR</code> package naming conventions.
<code>TwoSampleMR</code> provides the function
<code><a href="https://rdrr.io/pkg/TwoSampleMR/man/format_data.html" class="external-link">TwoSampleMR::format_data</a></code> to appropriately rename the
columns, as well as extra data checking and imputation processing. To
configure this correctly we need to provide a column mapping
<code>config.yml</code> file to the constructor of this package’s
<code>GWAS</code> class. An example of the <code>config.yml</code>
structure can be opened in <code>R</code> with the
<code><a href="../reference/get_default_gwas_config.html">get_default_gwas_config()</a></code> command - a copy of this should
be saved and adjusted as needed.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># File path to the config file; set open_file=TRUE to also open the file in RStudio</span></span>
<span><span class="va">fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_default_gwas_config.html">get_default_gwas_config</a></span><span class="op">(</span>open_file<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">fp</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/MRLipidInfection/extdata/config/default_gwas.yml"</span></span>
<span></span>
<span><span class="co"># Set the config file for the GLGC GWAS data</span></span>
<span><span class="va">config</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/config/glgc_gwas.yml"</span>, package <span class="op">=</span> <span class="st">"MRLipidInfection"</span><span class="op">)</span></span>
<span><span class="va">config</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/MRLipidInfection/extdata/config/glgc_gwas.yml"</span></span>
<span></span>
<span><span class="co"># View the structure of the default config file</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">"old_name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu">yaml</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yaml/man/read_yaml.html" class="external-link">read_yaml</a></span><span class="op">(</span><span class="va">fp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                        <span class="st">"new_name"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">yaml</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/yaml/man/read_yaml.html" class="external-link">read_yaml</a></span><span class="op">(</span><span class="va">fp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">old_name</th>
<th align="left">new_name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">exposure</td>
<td align="left">type</td>
</tr>
<tr class="even">
<td align="left">Phenotype</td>
<td align="left">phenotype_col</td>
</tr>
<tr class="odd">
<td align="left">SNP</td>
<td align="left">snp_col</td>
</tr>
<tr class="even">
<td align="left">beta</td>
<td align="left">beta_col</td>
</tr>
<tr class="odd">
<td align="left">se</td>
<td align="left">se_col</td>
</tr>
<tr class="even">
<td align="left">eaf</td>
<td align="left">eaf_col</td>
</tr>
<tr class="odd">
<td align="left">effect_allele</td>
<td align="left">effect_allele_col</td>
</tr>
<tr class="even">
<td align="left">other_allele</td>
<td align="left">other_allele_col</td>
</tr>
<tr class="odd">
<td align="left">pval</td>
<td align="left">pval_col</td>
</tr>
<tr class="even">
<td align="left">units</td>
<td align="left">units_col</td>
</tr>
<tr class="odd">
<td align="left">ncase</td>
<td align="left">ncase_col</td>
</tr>
<tr class="even">
<td align="left">ncontrol</td>
<td align="left">ncontrol_col</td>
</tr>
<tr class="odd">
<td align="left">samplesize</td>
<td align="left">samplesize_col</td>
</tr>
<tr class="even">
<td align="left">gene</td>
<td align="left">gene_col</td>
</tr>
<tr class="odd">
<td align="left">id</td>
<td align="left">id_col</td>
</tr>
<tr class="even">
<td align="left">z</td>
<td align="left">z_col</td>
</tr>
<tr class="odd">
<td align="left">info</td>
<td align="left">info_col</td>
</tr>
<tr class="even">
<td align="left">chr</td>
<td align="left">chr_col</td>
</tr>
<tr class="odd">
<td align="left">pos</td>
<td align="left">pos_col</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="exposure-data">Exposure data<a class="anchor" aria-label="anchor" href="#exposure-data"></a>
</h2>
<div class="section level3">
<h3 id="lipid-profile-data">Lipid profile data<a class="anchor" aria-label="anchor" href="#lipid-profile-data"></a>
</h3>
<p>Next we must obtain the data for the exposure. We are interested
whether exposure to lipid lowering medications influences risk of
developing sepsis. This exact data is not readily available on a
population level and has many confounding factors. We will therefore
estimate the effect of genetically proxied lipid medication therapy by
looking for Single Nucleotide Polymorphisms (SNPs) in close proximity to
known lipid metabolism genes (HMG-CoA Reductase, NPC1L1, PCSK9, CETP)
that are significantly associated with circulating lipid levels.</p>
<p>We will use the the latest Global Lipids Genetics Consortium <a href="http://csg.sph.umich.edu/willer/public/glgc-lipids2021/" class="external-link">Ancestry-specific
GWAS summary statistics</a> for the lipid components: HDL-C, LDL-C,
nonHDL-C, TC and TG.</p>
<p>As we will be using sepsis outcome data from the UK BioBank we will
use the data excluding the UK BioBank to avoid overlap of the exposure
and outcome populations.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List the URLs of the data sources</span></span>
<span><span class="va">urls_lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"LDL"</span>    <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_LDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  <span class="st">"HDL"</span>    <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_HDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  <span class="st">"TC"</span>     <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_TC_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  <span class="st">"nonHDL"</span> <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_nonHDL_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span>,</span>
<span>  <span class="st">"TG"</span>     <span class="op">=</span> <span class="st">"http://csg.sph.umich.edu/willer/public/glgc-lipids2021/results/ancestry_specific/without_UKB_logTG_INV_EUR_HRC_1KGP3_others_ALL.meta.singlevar.results.gz"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>We create a custom <code>GWAS</code> object for each lipid component.
Internally, the GWAS object reads in the data file provided through the
parameter <code>file_path</code>. If <code>file_path</code> is a URL
then the file is first downloaded into the package downloads directory
and extracted.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##Example GWAS object</span></span>
<span><span class="va">test_filepath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"test_data.csv"</span>,   package<span class="op">=</span><span class="st">"MRLipidInfection"</span><span class="op">)</span></span>
<span><span class="va">test_config</span>   <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"test_config.yml"</span>, package<span class="op">=</span><span class="st">"MRLipidInfection"</span><span class="op">)</span></span>
<span><span class="va">testGWAS</span>      <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWAS-class.html">GWAS</a></span><span class="op">(</span>file_path <span class="op">=</span> <span class="va">test_filepath</span>, </span>
<span>                      config    <span class="op">=</span> <span class="va">test_config</span>, </span>
<span>                      name      <span class="op">=</span> <span class="st">"test"</span>, </span>
<span>                      type      <span class="op">=</span> <span class="st">"exposure"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## GWAS object structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">testGWAS</span><span class="op">)</span></span>
<span><span class="co">#&gt; Formal class 'GWAS' [package "MRLipidInfection"] with 10 slots</span></span>
<span><span class="co">#&gt;   ..@ type     : chr "exposure"</span></span>
<span><span class="co">#&gt;   ..@ name     : chr "test"</span></span>
<span><span class="co">#&gt;   ..@ file_path: chr "/home/runner/work/_temp/Library/MRLipidInfection/testdata/test_data.csv"</span></span>
<span><span class="co">#&gt;   ..@ config   : chr "/home/runner/work/_temp/Library/MRLipidInfection/testdata/test_config.yml"</span></span>
<span><span class="co">#&gt;   ..@ overwrite: logi FALSE</span></span>
<span><span class="co">#&gt;   ..@ sig_pval : num 5e-08</span></span>
<span><span class="co">#&gt;   ..@ col_map  :List of 13</span></span>
<span><span class="co">#&gt;   .. ..$ type             : chr "exposure"</span></span>
<span><span class="co">#&gt;   .. ..$ phenotype_col    : chr "Phenotype"</span></span>
<span><span class="co">#&gt;   .. ..$ snp_col          : chr "SNP"</span></span>
<span><span class="co">#&gt;   .. ..$ beta_col         : chr "beta"</span></span>
<span><span class="co">#&gt;   .. ..$ se_col           : chr "se"</span></span>
<span><span class="co">#&gt;   .. ..$ eaf_col          : chr "eaf"</span></span>
<span><span class="co">#&gt;   .. ..$ effect_allele_col: chr "effect_allele"</span></span>
<span><span class="co">#&gt;   .. ..$ other_allele_col : chr "other_allele"</span></span>
<span><span class="co">#&gt;   .. ..$ pval_col         : chr "pval"</span></span>
<span><span class="co">#&gt;   .. ..$ samplesize_col   : chr "samplesize"</span></span>
<span><span class="co">#&gt;   .. ..$ id_col           : chr "id"</span></span>
<span><span class="co">#&gt;   .. ..$ chr_col          : chr "chr"</span></span>
<span><span class="co">#&gt;   .. ..$ pos_col          : chr "pos"</span></span>
<span><span class="co">#&gt;   ..@ data     :Classes 'data.table' and 'data.frame':   9 obs. of  15 variables:</span></span>
<span><span class="co">#&gt;   .. ..$ SNP                   : chr [1:9] "rs1123571" "rs903911" "rs6698680" "rs6685829" ...</span></span>
<span><span class="co">#&gt;   .. ..$ chr.exposure          : int [1:9] 1 1 1 1 1 1 1 1 1</span></span>
<span><span class="co">#&gt;   .. ..$ pos.exposure          : int [1:9] 2326009 2326812 2329661 2330016 2330190 2332391 2338126 2339139 2339427</span></span>
<span><span class="co">#&gt;   .. ..$ other_allele.exposure : chr [1:9] "G" "A" "A" "C" ...</span></span>
<span><span class="co">#&gt;   .. ..$ effect_allele.exposure: chr [1:9] "A" "G" "G" "T" ...</span></span>
<span><span class="co">#&gt;   .. ..$ samplesize.exposure   : int [1:9] 808038 810508 810507 808046 813372 821075 810363 787211 787211</span></span>
<span><span class="co">#&gt;   .. ..$ eaf.exposure          : num [1:9] 0.473 0.474 0.474 0.499 0.473 0.469 0.449 0.441 0.412</span></span>
<span><span class="co">#&gt;   .. ..$ beta.exposure         : num [1:9] -0.0111 -0.0108 -0.0107 -0.0106 -0.0109 ...</span></span>
<span><span class="co">#&gt;   .. ..$ se.exposure           : num [1:9] 0.00169 0.00169 0.00169 0.00169 0.00169 ...</span></span>
<span><span class="co">#&gt;   .. ..$ pval.exposure         : num [1:9] 5.26e-11 1.45e-10 2.32e-10 3.00e-10 1.13e-10 ...</span></span>
<span><span class="co">#&gt;   .. ..$ id.exposure           : chr [1:9] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;   .. ..$ exposure              : chr [1:9] "LDL" "LDL" "LDL" "LDL" ...</span></span>
<span><span class="co">#&gt;   .. ..$ mr_keep.exposure      : logi [1:9] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ pval_origin.exposure  : chr [1:9] "reported" "reported" "reported" "reported" ...</span></span>
<span><span class="co">#&gt;   .. ..$ gene.exposure         : chr [1:9] NA NA NA NA ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;   ..@ use_flag : logi [1:9] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   ..@ gen_reg  :Formal class 'GenomeRegion' [package "MRLipidInfection"] with 12 slots</span></span>
<span><span class="co">#&gt;   .. .. ..@ genome_wide: logi TRUE</span></span>
<span><span class="co">#&gt;   .. .. ..@ name       : chr NA</span></span>
<span><span class="co">#&gt;   .. .. ..@ assembly   : chr "GRCh37"</span></span>
<span><span class="co">#&gt;   .. .. ..@ chromosome : num NA</span></span>
<span><span class="co">#&gt;   .. .. ..@ start      : num NA</span></span>
<span><span class="co">#&gt;   .. .. ..@ end        : num NA</span></span>
<span><span class="co">#&gt;   .. .. ..@ cis_tol    : num 3e+05</span></span>
<span><span class="co">#&gt;   .. .. ..@ clump_r2   : num 0.001</span></span>
<span><span class="co">#&gt;   .. .. ..@ clump_kb   : num 10000</span></span>
<span><span class="co">#&gt;   .. .. ..@ clump_p1   : num 1</span></span>
<span><span class="co">#&gt;   .. .. ..@ clump_p2   : num 1</span></span>
<span><span class="co">#&gt;   .. .. ..@ pop        : chr "EUR"</span></span></code></pre></div>
<p><strong>Warning</strong> The GLGC GWAS data files are large and will
take some time to download. However, creating further <code>GWAS</code>
objects with the same <code>file_path</code> will result in much quicker
loading as the class first checks whether the file has already been
downloaded. There are more than 47 million SNPs in the datasets. We need
to filter the GWAS data SNPs associated with circulating lipid levels at
a statistical significance of <code>p&lt;5e-8</code>. This threshold is
passed to the <code>GWAS</code> object at instantiation. Internally, the
<code>GWAS</code> caches the filters SNPs to a .csv file within the
package structure. This is loaded instead if the command is re-run
(behaviour that can be overridden by passing <code>overwrite=TRUE</code>
to the <code>GWAS</code> constructor).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a list of GWAS objects using the GLGC data</span></span>
<span><span class="va">lipids_gwas_lst</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">urls_lst</span><span class="op">)</span>,</span>
<span>                               .y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">urls_lst</span><span class="op">)</span>, </span>
<span>                               .f <span class="op">=</span> <span class="op">~</span><span class="fu"><a href="../reference/GWAS-class.html">GWAS</a></span><span class="op">(</span>file_path <span class="op">=</span> <span class="va">.x</span>, </span>
<span>                                          name      <span class="op">=</span> <span class="va">.y</span>,</span>
<span>                                          config    <span class="op">=</span> <span class="va">config</span>,</span>
<span>                                          type      <span class="op">=</span> <span class="st">"exposure"</span>,</span>
<span>                                          sig_pval  <span class="op">=</span> <span class="fl">5e-8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cis-acting-lipid-metabolism-gene-snps">Cis-acting lipid metabolism gene SNPs<a class="anchor" aria-label="anchor" href="#cis-acting-lipid-metabolism-gene-snps"></a>
</h3>
<p>We will create a set of exposure SNPs that will be subset of the SNPs
that associate with the 5 lipid components: HDL-C, LDL-C, nonHDL-C, TC
and TG. This subset of SNPs will be in close proximity to the genes of
interest (HMG-CoA Reductase, NPC1L1, PCSK9, CETP). Data regarding the
positions of the genes is available <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=HMGCR" class="external-link">online</a>.</p>
<p>The latest lipid data from the GLGC uses the Genome Reference
Consortium Human Reference 37 (GRCh37) Assembly and so we must use this
when referring to the position of the genes.</p>
<p>First, we create a list of objects of class <code>GenomeRegion</code>
- this is a simple custom S4 class containing gene information. In
addition, we will look at all of the SNPs across the whole genome that
associate with lipid profile components (set
<code>genome_wide=TRUE</code>). Each <code>GWAS</code> object has an
internal slot for a <code>GenomeRegion</code> object from which it will
determine the appropriate SNP filtering (e.g. keeping only SNPs around a
certain genome region / gene).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The HMG-CoA-R, NPC1L1, and PCSK9 genes; set the 'cis' base tolerance here.</span></span>
<span><span class="va">genome_regions_lst</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/GenomeRegion-class.html">GenomeRegion</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"HMGCR"</span>,  genome_wide<span class="op">=</span><span class="cn">FALSE</span>, chromosome<span class="op">=</span><span class="fl">5</span>,  start<span class="op">=</span><span class="fl">74632993</span>, end<span class="op">=</span><span class="fl">74657941</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span>, clump_r2<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/GenomeRegion-class.html">GenomeRegion</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"NPC1L1"</span>, genome_wide<span class="op">=</span><span class="cn">FALSE</span>, chromosome<span class="op">=</span><span class="fl">7</span>,  start<span class="op">=</span><span class="fl">44552134</span>, end<span class="op">=</span><span class="fl">44580929</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span>, clump_r2<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/GenomeRegion-class.html">GenomeRegion</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"PCSK9"</span>,  genome_wide<span class="op">=</span><span class="cn">FALSE</span>, chromosome<span class="op">=</span><span class="fl">1</span>,  start<span class="op">=</span><span class="fl">55505221</span>, end<span class="op">=</span><span class="fl">55530525</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span>, clump_r2<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/GenomeRegion-class.html">GenomeRegion</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"CETP"</span>,   genome_wide<span class="op">=</span><span class="cn">FALSE</span>, chromosome<span class="op">=</span><span class="fl">16</span>, start<span class="op">=</span><span class="fl">56995862</span>, end<span class="op">=</span><span class="fl">57017757</span>, cis_tol<span class="op">=</span><span class="fl">1e5</span>, clump_r2<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/GenomeRegion-class.html">GenomeRegion</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"GenomeWide"</span>, genome_wide<span class="op">=</span><span class="cn">TRUE</span>, clump_r2<span class="op">=</span><span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Next we create exposure data for each lipid component
<code>GWAS</code> and each <code>GenomeRegion</code>. Hence there will
be n = <code>length(lipids_gwas_lst)</code> x
<code>length(genome_regions_list)</code> exposure datasets.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create all the permutations of lipid components (#5) and genome regions (#5) = #25 datasets</span></span>
<span><span class="va">lipid_gene_perms_lst</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span></span>
<span>  .x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">lipids_gwas_lst</span>,    times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genome_regions_lst</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  .y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">genome_regions_lst</span>, each <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lipids_gwas_lst</span><span class="op">)</span><span class="op">)</span>, </span>
<span>  .f <span class="op">=</span> <span class="op">~</span> <span class="fu">`genome_region&lt;-`</span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clumping-snps">Clumping SNPs<a class="anchor" aria-label="anchor" href="#clumping-snps"></a>
</h3>
<p>Many of the SNPs will be highly correlated, i.e. be genetically
linked and provide no additional information over and above the whole
set of correlated SNPs. To reduce the number of SNPs to only those most
significantly associated with lipid levels we use Linkage Disequilibrium
clumping. <a href="https://kp4cd.org/sites/default/files/documents/LD_clumping.pdf" class="external-link">Clumping</a>
reports the most significant genetic associations in a region in terms
of a smaller number of “clumps” of genetically linked SNPs.</p>
<p>SNPs that are correlated with an R<sup>2</sup> greater than some
value, <code>clump_r2</code>, will be clumped - i.e. the SNP with the
lower p-value will be removed. The process is iterative and has several
parameters: <code>clump_kb</code>, <code>clump_r2</code>,
<code>clump_p</code>. We will use the defaults, except for
<code>clump_r2</code> with the, which we will set more leniently,
instead of the default of 0.001, due to the fact we are looking at
cis-acting SNPs which will be more correlated by virtue of their
proximity to the genes of interest. Clumping parameters are passed in as
part of the <code>GenomeRegion</code> object (<em>above</em>).</p>
<p>We could do this on the Bristol IEU severs using the function
<code><a href="https://rdrr.io/pkg/TwoSampleMR/man/clump_data.html" class="external-link">TwoSampleMR::clump_data</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run clumping</span></span>
<span><span class="va">lipid_gene_perms_clump_lst</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">lipid_gene_perms_lst</span>, <span class="va">clump_snps</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_int</a></span><span class="op">(</span><span class="va">lipid_gene_perms_clump_lst</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">use_flag</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="outcome-data">Outcome data<a class="anchor" aria-label="anchor" href="#outcome-data"></a>
</h2>
<div class="section level3">
<h3 id="sepsis">Sepsis<a class="anchor" aria-label="anchor" href="#sepsis"></a>
</h3>
<p>The outcome we are interested in is sepsis. There are several GWAS
relating whether a patient develops sepsis to their underlying SNPs.
Several of these studies are available on the <a href="https://gwas.mrcieu.ac.uk" class="external-link">IEU GWAS database</a> which can be
queried using either the <code>TwoSampleMR</code> or
<code>ieugwasr</code> R packages.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># List of available GWAS summary statistics</span></span>
<span><span class="va">available_gwas</span> <span class="op">&lt;-</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ieugwasr/man/gwasinfo.html" class="external-link">gwasinfo</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="co"># Filter for those that mention 'sepsis' in the 'trait' description column</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"sepsis|pneumonia"</span>, <span class="va">.data</span><span class="op">$</span><span class="va">trait</span>, ignore.case<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># We will pick a few sepsis GWAS</span></span>
<span><span class="va">outcome_gwas_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ieu-b-5086"</span>, <span class="st">"ieu-b-5088"</span>, <span class="st">"ieu-b-4982"</span>, </span>
<span>                     <span class="st">"ieu-b-4980"</span>, <span class="st">"ieu-b-4981"</span>, <span class="st">"ieu-b-69"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extracting-outcome-data">Extracting outcome data<a class="anchor" aria-label="anchor" href="#extracting-outcome-data"></a>
</h3>
<p>Having identified the GWAS summary statistics we next need to extract
the data for our SNPs of interest (those that are highly predictive of
the exposure, in this case the components of the blood lipid
profile).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract data for cis-gene SNPs</span></span>
<span><span class="va">outcome_data</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">lipid_gene_perms_clump_lst</span>,</span>
<span>                           .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html" class="external-link">extract_outcome_data</a></span><span class="op">(</span></span>
<span>                              snps            <span class="op">=</span> <span class="va">.x</span><span class="op">@</span><span class="va">data</span><span class="op">$</span><span class="va">SNP</span><span class="op">[</span><span class="va">.x</span><span class="op">@</span><span class="va">use_flag</span><span class="op">]</span>,</span>
<span>                              outcomes        <span class="op">=</span> <span class="va">outcome_gwas_id</span>,</span>
<span>                              proxies         <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                              rsq             <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>                              align_alleles   <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                              palindromes     <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                              maf_threshold   <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>                              access_token    <span class="op">=</span> <span class="fu">ieugwasr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ieugwasr/man/check_access_token.html" class="external-link">check_access_token</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                              splitsize       <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>                              proxy_splitsize <span class="op">=</span> <span class="fl">500</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="harmonising">Harmonising<a class="anchor" aria-label="anchor" href="#harmonising"></a>
</h2>
<p>We have the effects of the (significant) SNPs on lipid levels and the
effect of these same SNPs on the outcome sepsis. However, it is
important to harmonise the effects. This means that the effect of a SNP
on the exposure and the effect of that SNP on the outcome must each
correspond to the same allele (i.e. the A&amp;Ts and C&amp;Gs cannot be
the wrong way round). <em>Note</em> The IEU GWAS database contains data
that is already harmonised. For more information on <a href="https://mrcieu.github.io/TwoSampleMR/articles/harmonise.html" class="external-link">harmonising</a>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter out the Gene:Lipid SNP associations with no significant SNPs</span></span>
<span><span class="va">zero_snps</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_lgl</a></span><span class="op">(</span><span class="va">lipid_gene_perms_clump_lst</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">use_flag</span><span class="op">)</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Harmonise the data for the LDL SNPs across the whole genome</span></span>
<span><span class="va">harm_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html" class="external-link">map2</a></span><span class="op">(</span>.x <span class="op">=</span> <span class="va">lipid_gene_perms_clump_lst</span><span class="op">[</span><span class="op">!</span><span class="va">zero_snps</span><span class="op">]</span>,</span>
<span>                        .y <span class="op">=</span> <span class="va">outcome_data</span><span class="op">[</span><span class="op">!</span><span class="va">zero_snps</span><span class="op">]</span>,</span>
<span>                        .f <span class="op">=</span> <span class="op">~</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html" class="external-link">harmonise_data</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">data</span>, <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level2">
<h2 id="two-sample-mendelian-randomisation">Two Sample Mendelian Randomisation<a class="anchor" aria-label="anchor" href="#two-sample-mendelian-randomisation"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The MR result for all of the different scenarios</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>res_dat <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="at">.x =</span> harm_dat, </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">.f =</span> <span class="sc">~</span> TwoSampleMR<span class="sc">::</span><span class="fu">mr</span>(.x)) <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up the name</span></span></code></pre></div>
<div class="section level3">
<h3 id="effects-and-significance">Effects and significance<a class="anchor" aria-label="anchor" href="#effects-and-significance"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Prepare the data</span></span>
<span><span class="va">plot_dat</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_df</a></span><span class="op">(</span><span class="va">res_dat</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">|&gt;</span> </span>
<span>                            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">id.exposure</span>, <span class="va">exposure</span>, <span class="va">outcome</span>, <span class="va">method</span>, <span class="va">b</span>, <span class="va">pval</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span> <span class="op">=</span> <span class="va">id.exposure</span>, <span class="st">"lipid"</span> <span class="op">=</span> <span class="va">exposure</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>                            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"lipid"</span>, <span class="st">"outcome"</span><span class="op">)</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Ensure present in the data, even if NA</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/complete.html" class="external-link">complete</a></span><span class="op">(</span><span class="va">gene</span>, <span class="va">lipid</span>, <span class="va">outcome</span>, <span class="va">method</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Just look at IVW for now</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">method</span> <span class="op">==</span> <span class="st">"Inverse variance weighted"</span>, <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Deselect to make table nicer</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">method</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="interpretation">Interpretation<a class="anchor" aria-label="anchor" href="#interpretation"></a>
</h3>
<p>Genetically proxied inhibition of important lipid metabolism drug
targets has no consistent effect on the risk of developing sepsis across
multiple cohorts. Apart from CETP which appeared to have significant
effects in 2 cohorts.</p>
<p>To be continued…</p>
<p><br></p>
</div>
</div>
<div class="section level2">
<h2 id="plots">Plots<a class="anchor" aria-label="anchor" href="#plots"></a>
</h2>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Set the significance cutoff</span></span>
<span><span class="va">p_val_cutoff</span> <span class="op">=</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="co"># Plot heat map of lipid component against lipid regulation genes</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plot_dat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">lipid</span>, y<span class="op">=</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># Set the fill tile colour to the p-value</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">pval</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="co"># Set the text to the beta value</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"\u03B2"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">b</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># Set the text to the p value</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"p ="</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">pval</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, nudge_y<span class="op">=</span><span class="op">-</span><span class="fl">0.15</span>, size<span class="op">=</span><span class="fl">3.5</span>, fontface <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># Ensure a very sharp colour transition at &lt;0.05 --&gt; green</span></span>
<span>    <span class="co"># Bottom 5% colour scale is green; top 95% colour scale is viridis gradient</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradientn</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"green"</span>,<span class="fl">50</span><span class="op">)</span>,</span>
<span>                                    <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/viridis_pal.html" class="external-link">viridis_pal</a></span><span class="op">(</span>option <span class="op">=</span> <span class="st">"mako"</span>, begin<span class="op">=</span><span class="fl">1</span>, end<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span><span class="op">(</span><span class="fl">950</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                         values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>, length.out<span class="op">=</span><span class="fl">1000</span><span class="op">)</span>,</span>
<span>                         <span class="co"># Set limits to ensure that default min(data)/max(data) is not used</span></span>
<span>                         limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                         breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.0</span>, <span class="fl">0.75</span>, <span class="fl">0.50</span>, <span class="fl">0.25</span>, <span class="va">p_val_cutoff</span><span class="op">)</span>,</span>
<span>                         labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/label_number.html" class="external-link">label_number</a></span><span class="op">(</span>accuracy <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title    <span class="op">=</span><span class="st">"MR estimate by lipid metabolism gene and lipid profile component"</span>, </span>
<span>         subtitle <span class="op">=</span> <span class="st">"Colour represents p-value, with bright green colour at cut-off p&lt;0.05"</span>,</span>
<span>         x        <span class="op">=</span><span class="st">"Lipid component"</span>, </span>
<span>         y        <span class="op">=</span> <span class="st">"Gene"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">outcome</span>, ncol <span class="op">=</span> <span class="fl">1</span>, scales<span class="op">=</span><span class="st">'free'</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/mr_heatmap.png" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="mr-scatter-plots">MR scatter plots<a class="anchor" aria-label="anchor" href="#mr-scatter-plots"></a>
</h3>
<p><img src="figures/mr_scatter_1.png"><img src="figures/mr_scatter_10.png"><img src="figures/mr_scatter_100.png"><img src="figures/mr_scatter_101.png"><img src="figures/mr_scatter_102.png"><img src="figures/mr_scatter_103.png"><img src="figures/mr_scatter_104.png"><img src="figures/mr_scatter_105.png"><img src="figures/mr_scatter_106.png"><img src="figures/mr_scatter_107.png"><img src="figures/mr_scatter_108.png"><img src="figures/mr_scatter_109.png"><img src="figures/mr_scatter_11.png"><img src="figures/mr_scatter_110.png"><img src="figures/mr_scatter_111.png"><img src="figures/mr_scatter_112.png"><img src="figures/mr_scatter_113.png"><img src="figures/mr_scatter_114.png"><img src="figures/mr_scatter_115.png"><img src="figures/mr_scatter_116.png"><img src="figures/mr_scatter_117.png"><img src="figures/mr_scatter_118.png"><img src="figures/mr_scatter_119.png"><img src="figures/mr_scatter_12.png"><img src="figures/mr_scatter_120.png"><img src="figures/mr_scatter_121.png"><img src="figures/mr_scatter_122.png"><img src="figures/mr_scatter_123.png"><img src="figures/mr_scatter_124.png"><img src="figures/mr_scatter_125.png"><img src="figures/mr_scatter_126.png"><img src="figures/mr_scatter_13.png"><img src="figures/mr_scatter_14.png"><img src="figures/mr_scatter_15.png"><img src="figures/mr_scatter_16.png"><img src="figures/mr_scatter_17.png"><img src="figures/mr_scatter_18.png"><img src="figures/mr_scatter_19.png"><img src="figures/mr_scatter_2.png"><img src="figures/mr_scatter_20.png"><img src="figures/mr_scatter_21.png"><img src="figures/mr_scatter_22.png"><img src="figures/mr_scatter_23.png"><img src="figures/mr_scatter_24.png"><img src="figures/mr_scatter_25.png"><img src="figures/mr_scatter_26.png"><img src="figures/mr_scatter_27.png"><img src="figures/mr_scatter_28.png"><img src="figures/mr_scatter_29.png"><img src="figures/mr_scatter_3.png"><img src="figures/mr_scatter_30.png"><img src="figures/mr_scatter_31.png"><img src="figures/mr_scatter_32.png"><img src="figures/mr_scatter_33.png"><img src="figures/mr_scatter_34.png"><img src="figures/mr_scatter_35.png"><img src="figures/mr_scatter_36.png"><img src="figures/mr_scatter_37.png"><img src="figures/mr_scatter_38.png"><img src="figures/mr_scatter_39.png"><img src="figures/mr_scatter_4.png"><img src="figures/mr_scatter_40.png"><img src="figures/mr_scatter_41.png"><img src="figures/mr_scatter_42.png"><img src="figures/mr_scatter_43.png"><img src="figures/mr_scatter_44.png"><img src="figures/mr_scatter_45.png"><img src="figures/mr_scatter_46.png"><img src="figures/mr_scatter_47.png"><img src="figures/mr_scatter_48.png"><img src="figures/mr_scatter_49.png"><img src="figures/mr_scatter_5.png"><img src="figures/mr_scatter_50.png"><img src="figures/mr_scatter_51.png"><img src="figures/mr_scatter_52.png"><img src="figures/mr_scatter_53.png"><img src="figures/mr_scatter_54.png"><img src="figures/mr_scatter_55.png"><img src="figures/mr_scatter_56.png"><img src="figures/mr_scatter_57.png"><img src="figures/mr_scatter_58.png"><img src="figures/mr_scatter_59.png"><img src="figures/mr_scatter_6.png"><img src="figures/mr_scatter_60.png"><img src="figures/mr_scatter_61.png"><img src="figures/mr_scatter_62.png"><img src="figures/mr_scatter_63.png"><img src="figures/mr_scatter_64.png"><img src="figures/mr_scatter_65.png"><img src="figures/mr_scatter_66.png"><img src="figures/mr_scatter_67.png"><img src="figures/mr_scatter_68.png"><img src="figures/mr_scatter_69.png"><img src="figures/mr_scatter_7.png"><img src="figures/mr_scatter_70.png"><img src="figures/mr_scatter_71.png"><img src="figures/mr_scatter_72.png"><img src="figures/mr_scatter_73.png"><img src="figures/mr_scatter_74.png"><img src="figures/mr_scatter_75.png"><img src="figures/mr_scatter_76.png"><img src="figures/mr_scatter_77.png"><img src="figures/mr_scatter_78.png"><img src="figures/mr_scatter_79.png"><img src="figures/mr_scatter_8.png"><img src="figures/mr_scatter_80.png"><img src="figures/mr_scatter_81.png"><img src="figures/mr_scatter_82.png"><img src="figures/mr_scatter_83.png"><img src="figures/mr_scatter_84.png"><img src="figures/mr_scatter_85.png"><img src="figures/mr_scatter_86.png"><img src="figures/mr_scatter_87.png"><img src="figures/mr_scatter_88.png"><img src="figures/mr_scatter_89.png"><img src="figures/mr_scatter_9.png"><img src="figures/mr_scatter_90.png"><img src="figures/mr_scatter_91.png"><img src="figures/mr_scatter_92.png"><img src="figures/mr_scatter_93.png"><img src="figures/mr_scatter_94.png"><img src="figures/mr_scatter_95.png"><img src="figures/mr_scatter_96.png"><img src="figures/mr_scatter_97.png"><img src="figures/mr_scatter_98.png"><img src="figures/mr_scatter_99.png"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicholas Sunderland.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
